### PR TITLE
InitChain message gets send at genesis.

### PR DESCRIPTION
**Pull request by ebuchman**
Openning PR for review

### CODE REVIEW

**Comment by ebuchman** [comment]
Interesting. I was thinking of putting this in the node/node.go but maybe this is better. I think we can get away with just one call tho - why don't we just add a single `if appHeight == 0` and send InitChain there, before we start the rest of the logic ?

**Comment by adrianbrink** [comment]
I thought about node.go as well, but it felt more natural here.

I pulled it out of the conditional blocks and removed some old comments.

**Comment by ebuchman** [comment]
Great. Also, let's move the `covertValidators` function to here: https://github.com/tendermint/tendermint/blob/master/types/protobuf.go

File: consensus/replay.go [hunk 1]
```diff
@@ -257,6 +257,13 @@ func (h *Handshaker) ReplayBlocks(appHash []byte, appBlockHeight int, proxyApp p
 	stateBlockHeight := h.state.LastBlockHeight
 	log.Notice("ABCI Replay Blocks", "appHeight", appBlockHeight, "storeHeight", storeBlockHeight, "stateHeight", stateBlockHeight)
 
+	// If appBlockHeight == 0 it means that we are at genesis and hence should send InitChain
+	// TODO: Currently we have 2 separate types of Validators (abci and types) and hence we need the conversion.
```
Comment (ebuchman) [inline consensus/replay.go]: no need for this TODO

**Comment by ebuchman** [comment]
Ok. remove that TODO and squash down to one commit and we're good :)

File: consensus/replay.go (hunk 1 above)
Comment (adrianbrink) [inline consensus/replay.go]: Good catch

**Comment by adrianbrink** [comment]
@ebuchman It fails, but it still seems cryptic to me. The TestHandshakeReplayAll test fails.

**Comment by adrianbrink** [comment]
@ebuchman The tests are passing locally inside the vagrant box. I've pushed again to see if they pass CircleCI.

I think the problem for the inconsistent test failures was that the VM didn't have enough memory. How much memory is available in CircleCI?

**Comment by melekes** [comment]
@adrianbrink note that CircleCI does not use our Vagrantfile, so setting higher memory for it won't help.

**Comment by ebuchman** [comment]
I believe the issue was due to some hard coded test data that is changed by InitChain, since calling InitChain actually updates the app state, this changes the app root hash

**Comment by adrianbrink** [comment]
@melekes Yes, but I think that's the reason why my test failed locally in the beginning. Plus the fact that some of the test data was incorrect.

@ebuchman That would be nice. You did update the tests in 495283e, correct?

**Comment by ebuchman** [comment]
Yep

**Pull request merged**